albatross/covariance_functions/covariance_function.h

```diff
@@ -136,6 +136,16 @@ class CovarianceFunction : public ParameterHandlingMixin {
     return derived().call_impl_(y, x);
   }
 
+  /*
+   * Covariance between each element and every other in a vector.
+   */
+  template <typename X,
+            typename std::enable_if<
+                has_defined_call_impl<Derived, X &, X &>::value, int>::type = 0>
+  double operator()(const X &x) const {
```

Comment:
doesn't `Derived` need to have a `defined_call_impl` with `const X &` here?

Reply:
I would think so.
I also think there's a syntax that would allow not repeating the type which would prevent this potential issue, but it'd be kinda weird looking. Something like this:
```
template<typename X>
auto operator()(const X &x) const -> std::enable_if<has_defined_call_impl<Derived, decltype(x), decltype(x)>::value, double>::type;
```

Reply:
Yeah good point.  I do think we want all arguments to be const references.

@jbangelo I think "kinda weird looking" is a good way to describe albatross.  One alternative (or maybe supplement) could be to change `has_defined_call_impl<X, T>` to only check for `X::call_impl_(const T &)`.